robot/EDA/resources/TriggerHandlersPageObject.py

```diff
@@ -0,0 +1,195 @@
+        self.selenium.wait_until_page_contains_element(
+            locator,
+            error="Action '" + action + "' doesn't exist in the list of Action Shortcuts"
+        )
+        self.selenium.click_element(locator)
+
+    def select_EDA_tab(self, tab):
+        """ Selects Contacts tab - as passed in to the function """
+        locator_menu = trigger_handlers_locators["tab_menu"].format(tab)
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", locator_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_menu):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def verify_trigger_handler(self, trigger):
+        """ Verify that trigger handler exists """
+
+        locator_trigger = trigger_handlers_locators["trigger"].format(trigger)
+
+        self.selenium.wait_until_page_contains_element(
+            trigger_handlers_locators["recently_viewed"],
+            error=locator_trigger + " locator is not available in Trigger Handlers page"
+        )
+        self.selenium.click_element(trigger_handlers_locators["recently_viewed"])
+
+        self.selenium.wait_until_page_contains_element(
+            trigger_handlers_locators["all_main_fields"],
+            error=trigger_handlers_locators["all_main_fields"] + " locator is not available in Trigger Handlers page"
+        )
+        self.selenium.click_element(trigger_handlers_locators["all_main_fields"])
+
+        self.selenium.wait_until_page_contains_element(
+            trigger_handlers_locators["search_exists"],
+            error=trigger + " object not found in Trigger Handlers"
+        )
+        self.selenium.click_element(trigger_handlers_locators["search_exists"])
+        self.selenium.get_webelement(trigger_handlers_locators["search_exists"]).send_keys(trigger + Keys.RETURN)
+
+        # Verify that the CON_PreferredPhone_TDTM object has been found
+        self.selenium.wait_until_page_contains_element(
+            trigger_handlers_locators["preferred_phone"],
+            error="CON_PreferredPhone_TDTM not found in Trigger Handler list"
+        )
+
+        time.sleep(2)
+        self.selenium.wait_until_page_contains_element(
+            trigger_handlers_locators["trigger_handler"],
+            error="Unable to find Preferred Phone Trigger Handler "
+        )
+        self.selenium.click_element(trigger_handlers_locators["trigger_handler"])
+
+    def select_active_checkbox(self, loc_check, loc_checkbox):
```

Comment:
👍 I like how this keyword and the next are smart enough to check whether the item is already selected, and just returns if it's already selected.

Reply:
agreed! :-)

Reply:
I removed this file.